### DISCUSSION TITLE
Interfacing SOFA with ROS

### DISCUSSION THREAD

**Discussion by HoangGiang93** [sofa-framework/sofa#3885] [Share your achievements]
This [video](https://youtu.be/MoYzISmLexQ) showcases how to interface SOFA with ROS using Python and visualize the scene using Qt or GLFW. Below, you can find the displayed outcome:

https://github.com/sofa-framework/sofa/assets/64316740/82cb7113-7e92-4686-867b-36e9f31b1fbe

Prerequisites:
- [SOFA v22.12.00](https://github.com/sofa-framework/sofa/releases/tag/v22.12.00) with PLUGIN_SOFAPYTHON3 enabled
- [ROS](http://wiki.ros.org/noetic)

**Comment by manas-narayan** [comment]
> This [video](https://youtu.be/MoYzISmLexQ) showcases how to interface SOFA with ROS using Python and visualize the scene using Qt or GLFW. Below, you can find the displayed outcome:
> 
>  sofa_ros_result.mp4 
> Prerequisites:
> 
> * [SOFA v22.12.00](https://github.com/sofa-framework/sofa/releases/tag/v22.12.00) with PLUGIN_SOFAPYTHON3 enabled
> * [ROS](http://wiki.ros.org/noetic)

Hey Hoang,
Can you guide me how you interfaced both SOFA and ROS together. I have ROS noetic installed and sofa v23. This is for my project. Any help will be very much appreciated.
Thank you

**Reply by HoangGiang93** [reply]
Thank you for your interest. Yes, if you click on the [video](https://youtu.be/MoYzISmLexQ), you would see how to interface SOFA with ROS using Python and visualize the scene using Qt or GLFW.
